### PR TITLE
Added abstract class for freertos task creation

### PR DESCRIPTION
**Pull request by g-lesssard**
Simple class with a wrapper to objectify task creation. I tried adding tests but I had multiple issues starting and stopping the task scheduler and they didn't add much value.

### CODE REVIEW

File: src/taskManager/include/taskManager/AbstractTask.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include <algorithm>
```
Comment (cquesnel) [inline src/taskManager/include/taskManager/AbstractTask.h:1]: The include should be after the `ifndef`

**Review by cquesnel — CHANGES_REQUESTED** [review]

**Review by xgroleau — COMMENTED** [review]
Two small comments

File: src/taskManager/src/AbstractTask.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    m_stackDepth = stackDepth;
+    m_priority = priority;
+    m_taskHandle = NULL;
+}
+
+void AbstractTask::start() {
+    xTaskCreate(wrapper, m_taskName, m_stackDepth, this, m_priority, &this->m_taskHandle);
+}
+
+TaskHandle_t AbstractTask::getTaskHandle() const { return m_taskHandle; }
```
[the remark this answers is not in the document]
Comment (g-lesssard) [reply src/taskManager/src/AbstractTask.cpp:15]: The task handle is used to get the status of the task. It is set to null in the constructor but is changed upon task creation (xTaskCreate).

File: src/taskManager/include/taskManager/AbstractTask.h (hunk 1 above)
Comment (g-lesssard) [reply src/taskManager/include/taskManager/AbstractTask.h]: Fixed

File: src/taskManager/src/AbstractTask.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+AbstractTask::AbstractTask(const char* taskName, portSHORT stackDepth, UBaseType_t priority) {
+    m_taskName = taskName;
+    m_stackDepth = stackDepth;
+    m_priority = priority;
+    m_taskHandle = NULL;
+}
+
+void AbstractTask::start() {
+    xTaskCreate(wrapper, m_taskName, m_stackDepth, this, m_priority, &this->m_taskHandle);
```
[the remark this answers is not in the document]
Comment (g-lesssard) [reply src/taskManager/src/AbstractTask.cpp]: Removed

**Review by xgroleau — APPROVED** [review]
GJ, we may migrate to a static allocation using xTaskCreateStatic eventually, maybe a template for param for the stack size? Just something to think about in the meantime. anyway we will check once we will be there 99% of the code will be reused, we will probably just add an std::arrray

**Review by cquesnel — APPROVED** [review]

**Pull request merged**